packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/VXZoom/VXZoom.js

```diff
@@ -62,6 +69,67 @@ class VXZoom extends PureComponent {
     }
   }
 
+  onPan (direction) {
+    const { 
+      transformMatrix: {
+        translateX,
+        translateY
+      }
+    } = this.zoom
+    const panDistance = 20
+    const newTransformation = { 
+      down: { translateX, translateY: translateY - panDistance },
+      left: { translateX: translateX + panDistance, translateY },
+      right: { translateX: translateX - panDistance, translateY },
+      up: { translateX, translateY: translateY + panDistance }
+    }
+    this.zoom.setTranslate(newTransformation[direction])
+  }
+
+  onKeyDown (event) {
+    const { panning, zooming } = this.props
+    const htmlTag = event.target?.tagName.toLowerCase()
+    if (htmlTag === 'rect' && (zooming || panning)) {
+      switch (event.key) {
+        case '+':
+        case '=': {
+          this.zoomIn()
+          return true
+        }
+        case '-':
+        case '_': {
+          this.zoomOut()
+          return true
+        }
+        case 'ArrowRight': {
+          event.preventDefault()
+          this.onPan('right')
+          return true
+        }
+        case 'ArrowLeft': {
+          event.preventDefault()
+          this.onPan('left')
+          return true
+        }
+        case 'ArrowUp': {
+          event.preventDefault()
+          this.onPan('up')
+          return true
+        }
+        case 'ArrowDown': {
+          event.preventDefault()
+          this.onPan('down')
+          return true
+        }
+        default: {
+          return true
+        }
+      }
+    }
```

Comment:
This function is almost identical to the keydown handler in `withKeyZoom`. Is it possible to merge them and decorate this component with `withKeyZoom(VXZoom)` to add the `onKeyDown` handler?

Reply:
It was difficult to get all three zoom/pan methods to work together. I'd prefer we circle back to RFC #1645 and consolidate on a single approach to panning rather than support panning using viewport and panning using a transformation matrix simultaneously by a single method. I'd like to merge this as is and experiment with swapping the `SingleImageViewer` over to the transformation matrix for pan and zoom instead. Is that ok?

Reply:
Sure, but `withKeyZoom` is a decorator that maps keys to callback functions. I don't think it's tied to a particular implementation of zoom. It's used by both the light curve viewer and single image viewer, which have very different implementations of pan and zoom.

Reply:
The image toolbar uses `withKeyZoom` to connect to the subject viewer store and control the scatter plot viewer from the keyboard when the toolbar has focus.
https://github.com/zooniverse/front-end-monorepo/blob/9f17eea42d4c9926731a81fd0496bbd9dab73122/packages/lib-classifier/src/components/Classifier/components/ImageToolbar/ImageToolbar.js#L14-L15

I've checked in the storybook for this PR. When the toolbar has focus, in Chrome and in Firefox, I can control the scatter plot viewer from the keyboard.

Sorry, but I'm going to ask for changing the keyboard handling here to use `withKeyZoom` since it's already working with the toolbar and I'm not following why the code needs to be implemented in two places.

Reply:
I think I got confused about all the implementations and keeping track how this all functioned. I'll have another look, but I still want us to consolidate on a single method.

Reply:
Ok, this is refactored to use the existing `onKeyDown` handler. I think some documentation would have been helpful here. Would you have time to add some for `onKeyDown`? I can document `VXZoom` in the future, but it'll have to be after I finish up on the VSLCV / upcoming DataImageViewer.

Reply:
Sure. The `withKeyZoom` component was written while we were rushing to get PH-TESS launched, so documenting it kind of got lost along the way, but it's a lightweight component. I think a short Readme should do. From memory, it connects a component up to the `SubjectViewer` store so that keyboard events on that component will fire the store's `onPan` and `onZoom` actions. (I can't remember the exact names of the actions.)

Reply:
#1726 adds a Readme and some new actions in the subject viewer store.

Reply:
Thank you!